liferay-frontend/liferay-portal
# LPS-131906 batch bug when registering an extended entity

@drewbrokke pinged me because when defining an API that re-uses an existing entity and also has Batch enabled, batch registration fails because it tries to register the same entity twice. So he has `UserAccount` API in `headless-admin-taxonomy` and `AccountUser` API that *reuses `UserAccount`* entity (and uses the extended proccess to add properties) in `account-rest`.

Batch has a completely undocumented way of supporting this use case: `batch.engine.task.item.delegate.name`. If you provide a value to that property it can register the same entity twice but noone is propagating the value to the delegate. Also casting fails for query parameter. This PR fixes both issues.

To be able to set `batch.engine.task.item.delegate.name` you have to add and deploy an OSGI property file that targets the desired resource (`UserAccountResourceImpl` in this case) and fix some wrong `operationId` that were not removed or were hardcoded (`postUserAccount` should be `postAccountUserAccount`).

To test it you can do a CURL request like this:

```
curl -X "POST" "http://localhost:8080/o/headless-batch-engine/v1.0/import-task/com.liferay.headless.admin.user.dto.v1_0.UserAccount?taskItemDelegateName=AccountUser&accountId=63402" \
     -H 'Content-Type: application/json' \
     -u 'test@liferay.com:test' \
     -d $'[
  {
    "emailAddress": "1@a.com",
    "givenName": "3",
    "familyName": "2",
    "alternateName": "4"
  }
]'
```

Notice that you have to pass taskItemDelegateName=AccountUser & accountId=63402. First one to identify the entity (DEFAULT is for UserAccount in Headless Admin User) and the second one because path parameters are query parameters in batch.

Then you can query the state in 

```
curl "http://localhost:8080/o/headless-batch-engine/v1.0/import-task/BATCH_ID" \
     -H 'Accept: application/json' \
     -u 'test@liferay.com:test'
```

This PR won't pass until https://github.com/liferay-frontend/liferay-portal/pull/1036 is merged, so ci:stop-ping for now.
